src/metabase/lib/metric.cljc

```diff
@@ -136,14 +137,19 @@
     mbql.normalize/normalize))
 
 (defmethod lib.metadata.calculation/metadata-method :metric
-  [query stage-number [_ _ metric-id]]
-  (let [metric-meta (lib.metadata/metric query metric-id)
-        metric-aggregation (some-> metric-meta
-                                   :dataset-query
-                                   normalize-legacy-query
-                                   lib.convert/->pMBQL
-                                   lib.aggregation/aggregations
-                                   first)
-        metric-name (:name metric-meta)]
-    (assoc (lib.metadata.calculation/metadata query stage-number metric-aggregation)
-           :display-name metric-name)))
+  [query _stage-number [_ opts metric-id]]
+  (let [metric-meta       (lib.metadata/metric query metric-id)
+        metric-query      (lib.query/query query (normalize-legacy-query (:dataset-query metric-meta)))
+        inner-aggregation (first (lib.aggregation/aggregations metric-query))
+        inner-meta        (lib.metadata.calculation/metadata metric-query -1 inner-aggregation)]
+    (-> inner-meta
+        (assoc :display-name           (:name metric-meta) ; Metric card's name
+               :lib/hack-original-name (:name metric-meta) ; Metric card's name
+               :name                   (:name inner-meta)  ; Name of the inner aggregation column
+               :ident                  (:ident opts))      ; Ident of the `[:metric ...]` reference!
```

Comment:
Will this not be overridden right away?

Reply:
Yeah, the `u/assoc-dissoc` below does the right thing and this line is redundant.